crates/stages/src/stages/execution.rs

```diff
@@ -360,14 +345,34 @@ impl<DB: Database> Stage<DB> for ExecutionStage {
 
         for (key, storage) in storage_changeset_batch.into_iter().rev() {
             let address = key.address();
-            if plain_storage_cursor.seek_by_key_subkey(address, storage.key)?.is_some() {
-                plain_storage_cursor.delete_current()?;
+            if let Some(v) = plain_storage_cursor.seek_by_key_subkey(address, storage.key)? {
+                if v.key == storage.key {
```

Comment:
nit, is this needed? won't it always be hit by the cursor action?

Reply:
it is. as it was before it was a bug.

seek will get the key or greater. there was a situation where he was searching for `0x01`, but was getting `0x03`. After, we're deleting the `current()`, but since the cursor now is at `0x03` we're deleting the wrong one.